app/controllers/app.py

```diff
@@ -1224,3 +1224,70 @@ def post(self, project_id, app_id):
             return dict(status='fail', message='No values found'), 404
 
         return dict(status='success', data=dict(values=cpu_data_list)), 200
+
+
+class AppNetworkUsageView(Resource):
+    @jwt_required
+    def post(self, project_id, app_id):
+
+        current_user_id = get_jwt_identity()
+        current_user_roles = get_jwt_claims()['roles']
+
+        app_memory_schema = MetricsSchema()
+        app_network_data = request.get_json()
+
+        validated_query_data, errors = app_memory_schema.load(app_network_data)
+
+        if errors:
+            return dict(status='fail', message=errors), 400
+
+        project = Project.get_by_id(project_id)
+
+        if not project:
+            return dict(
+                status='fail',
+                message=f'project {project_id} not found'
+            ), 404
+
+        if not is_owner_or_admin(project, current_user_id, current_user_roles):
+            return dict(status='fail', message='unauthorised'), 403
+
+        # Check app from db
+        app = App.get_by_id(app_id)
+
+        if not app:
+            return dict(
+                status='fail',
+                message=f'app {app_id} not found'
+            ), 404
+
+        # Get current time
+        current_time = datetime.datetime.now()
+        yesterday = current_time + datetime.timedelta(days=-1)
+        namespace = project.alias
+        app_alias = app.alias
+
+        prometheus = Prometheus()
+
+        start = validated_query_data.get('start', yesterday.timestamp())
+        end = validated_query_data.get('end', current_time.timestamp())
+        step = validated_query_data.get('step', '1h')
+
+        prom_data = prometheus.query_rang(
+            start=start,
+            end=end,
+            step=step,
+            metric='sum(rate(container_network_receive_bytes_total{namespace="' +
```

Comment:
@Mubangizi shouldn't we include the bit of `container!="POD"` in the query to eliminate some repetition of the containers

Reply:
Nope all container labels have POD, and all containers have an image tag too, so no need

Reply:
How is this different from other metric endpoints?

Reply:
This makes sense after the testing